src/main/java/games/strategy/triplea/ui/TimespanDialog.java

```diff
@@ -0,0 +1,86 @@
+package games.strategy.triplea.ui;
+
+import java.awt.Component;
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.time.ZoneOffset;
+import java.time.temporal.ChronoUnit;
+import java.util.Date;
+import java.util.function.Consumer;
+import java.util.function.Function;
+
+import javax.swing.JComboBox;
+import javax.swing.JOptionPane;
+import javax.swing.JSpinner;
+import javax.swing.SpinnerNumberModel;
+import javax.swing.border.EmptyBorder;
+
+import com.google.common.annotations.VisibleForTesting;
+
+import swinglib.JLabelBuilder;
+import swinglib.JPanelBuilder;
+
+/**
+ * A UI-Utility class that can be used to prompt the user for a ban or mute time.
+ */
+public class TimespanDialog {
+
+  /**
+   * The possible time units and corresponding mappings.
+   */
+  @VisibleForTesting
+  enum TimeUnit {
+    MINUTES("Minutes", i -> Instant.now().plus(i, ChronoUnit.MINUTES)),
+    HOURS("Hours", i -> Instant.now().plus(i, ChronoUnit.HOURS)),
+    DAYS("Days", i -> Instant.now().plus(i, ChronoUnit.DAYS)),
+    WEEKS("Weeks", i -> LocalDateTime.now(ZoneOffset.UTC).plus(i, ChronoUnit.WEEKS).toInstant(ZoneOffset.UTC)),
+    MONTHS("Months", i -> LocalDateTime.now(ZoneOffset.UTC).plus(i, ChronoUnit.MONTHS).toInstant(ZoneOffset.UTC)),
+    YEARS("Years", i -> LocalDateTime.now(ZoneOffset.UTC).plus(i, ChronoUnit.YEARS).toInstant(ZoneOffset.UTC)),
+    FOREVER("Forever", i -> null);
+
+    private final String name;
+    private final Function<Integer, Instant> function;
+
+    private TimeUnit(final String name, final Function<Integer, Instant> supplier) {
+      this.name = name;
+      this.function = supplier;
+    }
+
+    @Override
+    public String toString() {
+      return name;
+    }
+
+    @VisibleForTesting
```

Comment:
Is this annotation needed?  This method is called from `prompt()`, so it's not really visible just for test purposes.  (Obviously, it could be called from `prompt()` even if it were marked private, but that smells a bit as the compiler will still generate a synthetic package-private pass-through method so the private method can be called at runtime.)

Reply:
From the javadocs of `@VisibleForTesting`:
> Annotates a program element that exists, or is more widely visible than otherwise necessary, only for use in test code.

If I mark this method as private, I can still call it from TimespanDialog, but no other classes, I just raised the visibility to be able to call this methods from tests.
So in my opinion this annotation is 100% appropriate

Reply:
I agree, you _can_ call it from `TimespanDialog` if it's private.  However, this is considered bad practice in some cases because the compiler actually generates a second package-private method (a so-called "synthetic accessor") that simply passes through to the private method.  There's even a warning you can turn on in most static analyzers for that (in Eclipse, it's under **Code Style > Access to a non-accessible member of an enclosing type**).

NBD though.  It just caught my attention because I expected a private method to be called by the class that declared it.  Otherwise, if it's being called by other types (including enclosing types), then that seems to imply the method is part of the class API.